### PR TITLE
Docs: Document TCP routes in Kubernetes

### PR DESCRIPTION
**Pull request by alexfornuto**
## Summary

- Documents the annotations to create a TCP route.
- Includes an example route

## Related issues

- Documents https://github.com/pomerium/ingress-controller/pull/196
- Resolves https://github.com/pomerium/internal/issues/840
- Depends on https://github.com/pomerium/pomerium/pull/3340 to avoid merge conflicts.

## Checklist

- [x] reference any related issues
- [x] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: docs/docs/k8s/ingress.md [hunk 1]
```diff
@@ -350,6 +352,41 @@ spec:
     secretName: example-tls
 ```
 
+## TCP Endpoints
+
+The example route below defines a route providing a [tunneled TCP connection](/docs/tcp/readme.md) to an upstream service listening for non-web traffic:
```
Comment (wasaga) [inline docs/docs/k8s/ingress.md:357]: maybe `non-HTTP` ?

File: docs/docs/k8s/ingress.md [hunk 2]
```diff
@@ -350,6 +352,41 @@ spec:
...
+                name: tcp-service
+                port:
+                  name: app
+```
+
+The important points to note in this example:
+
+- The annotation `ingress.pomerium.io/tcp_upstream:` is set to `"true"`,
+- `spec.rules.[].http.paths.[].path` is omitted,
+- `spec.rules.[].http.paths.[].pathType` is set to `ImplementationSpecific`,
```
Comment (wasaga) [inline docs/docs/k8s/ingress.md:384]: do we need change `is` into `should` ? 
cause all three conditions should be met for the route to be considered a TCP route.

**Review by wasaga — APPROVED** [review]

File: docs/docs/k8s/ingress.md (hunk 1 above)
Comment (alexfornuto) [reply docs/docs/k8s/ingress.md:357]: Why?

File: docs/docs/k8s/ingress.md (hunk 2 above)
Comment (alexfornuto) [reply docs/docs/k8s/ingress.md:384]: I see where you're coming from, but since we're referring to the specific example above, these things is/are.

**Pull request merged**